### PR TITLE
add a couple of examples of creating interceptors

### PR DESCRIPTION
**Pull request by cap10morgan**
I had a hard time finding an example of the (slightly changed in 0.4.0) interceptor create syntax, so I thought I'd add it here.

Happy to move it elsewhere, improve it, etc.

### CODE REVIEW

File: guides/documentation/service-interceptors.md [hunk 1]
```diff
@@ -149,6 +149,39 @@ functions of further interceptors in the path are invoked as if no
...
+  (:require [io.pedestal.interceptor :refer [interceptor]]))
+
+(def my-before-interceptor
+  (interceptor
+   {:name ::hello-world
+    :enter
+    (fn [context]
+      (assoc context :response 
+                     {:status 200 :body "Hello world!" 
+                      :headers {"Content-Type" "text/plain"}}))}))
```
Comment (jcf) [inline guides/documentation/service-interceptors.md]: Hi @cap10morgan,

We've used the built-in helpers a lot in our codebase for defining interceptors. I wonder if it'd be helpful to others if we mention these functions and macros?

``` clj
(require '[io.pedestal.interceptor.helpers :as interceptor])

(def my-before-interceptor
  (interceptor/before
    ::my-before-interceptor
    (fn [context]
      (assoc-in context [:request :example] true))))
```

Although I've avoided use of the macro forms to date, there are helpers that call `def` for you too. This snippet is copied out of a docstring in the helpers namespace:

``` clj
(defbefore parse-body-as-wibblefish
  [context]
  (let [request (:request context)
        new-request (assoc request :wibblefish-params
                           (wibblefish-parse (:body request)))]
    (assoc context :request new-request)))
```

https://github.com/pedestal/pedestal/blob/c6e4a1f7e9c411cc83305ddcc451a954ab931ddc/service/src/io/pedestal/interceptor/helpers.clj

Comment (ohpauleez) [inline guides/documentation/service-interceptors.md]: @jcf The macros are discouraged moving forward, and one should prefer using the Interceptor API directly where possible.

Comment (jcf) [inline guides/documentation/service-interceptors.md]: By "Interceptor API" does than mean `io.pedestal.http/interceptor` alone, or does that include the helper functions in `io.pedestal.interceptor.helpers`? We make pretty extensive use of the `before` function in our codebase, so I'd be sad to see it go.

Comment (ohpauleez) [inline guides/documentation/service-interceptors.md]: It means using `io.pedestal.interceptor/interceptor` directly - which has been overhauled.  It's simpler, better defined, and considerably less surface area.  The macros aren't going anywhere (they have a dedicated namespace), but they are discouraged and will break AOT compilation.

**Comment by ohpauleez** [comment]
Thanks Wes!  Sorry it took me so long to get to these.  If you ever need help with anything Pedestal related, feel free to tweet or email me directly!

**Comment by cap10morgan** [comment]
Cool, thanks @ohpauleez!

**Pull request merged**